app/_hub/kong-inc/statsd-advanced/index.md

```diff
@@ -43,36 +49,52 @@ params:
       required:
```

Comment:
> One question that I have about these empty `required` fields. In the schema, if it's left empty, it defaults to `false`. But in the docs, if left empty, it doesn't show anything of info to the user (as seen in the deploy preview). Is it okay to assume that users will understand that no `required` info means that it's `false` by default? I think the answer is _no_ because most of the plugins I've seen so far explicitly list their `required` fields (for example, [this](https://docs.konghq.com/hub/kong-inc/rate-limiting-advanced/)), so we probably want to do the same for this plugin too.

I had a convo with @bungle regarding this for some transformer plugins, and he mentioned this:

> bungle  9:47 AM
I don’t think we added that… I think in general fields that have default should be required if they for no specific reason also want to implement null
e.g. boolean  field with true or false should be enough, and null just confuses up
I will probably soon PR Kong with marking those fields that have default with also required=true

Reply:
@murillopaula I will mark all fields as required. Note that this makes for very large autogenerated examples from the template but so be it.

Reply:
@tlblessing no this is not correct, I can successfully create an instance of the plugin without entering any config field. When it doesn't have the explicit `required` in a top level field, it is really not required. We need to mark all the top level ones such as `host`, `port` and `metrics` as `required = false` instead.

Reply:
> @tlblessing no this is not correct, I can successfully create an instance of the plugin without entering any config field, meaning that the plugin does not have any required top level field. When it doesn't have the explicit `required` in a top level field, it is really not required. We need to mark all the top level ones such as `host`, `port` and `metrics` as `required = false` instead.

@murillopaula so host and port are listed as required for TCP and UDP. Those PRs were approved by you. So are those incorrect? It still seems to me that the fields are required but are autofilled with the default. The typedefs schema for host and port is not all that clear to me.

Reply:
@tlblessing ok, now I get it. The `default` fields are indeed auto filled with the default values, so they are indeed required.

Reply:
my bad, you are correct. all the fields of this plugins have default values 🤦 so we should mark them as required, which is what you're doing right now. sorry about the confusion